src/internal/storage/fileset/index/cache.go

```diff
@@ -0,0 +1,103 @@
+package index
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"sort"
+	"sync"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pbutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/chunk"
+)
+
+type Cache struct {
+	storage *chunk.Storage
+	cache   *simplelru.LRU
+	mu      sync.Mutex
+}
+
+func NewCache(storage *chunk.Storage, size int) *Cache {
+	lruCache, err := simplelru.NewLRU(size, nil)
+	if err != nil {
+		panic(err)
+	}
+	return &Cache{
+		storage: storage,
+		cache:   lruCache,
+	}
+}
+
+func (c *Cache) Get(ctx context.Context, chunkRef *chunk.DataRef, filter *pathFilter, w io.Writer) error {
+	c.mu.Lock()
+	v, ok := c.cache.Get(string(chunkRef.Ref.Id))
```

Comment:
this would break if the same chunk is ever looked up with two offsets, right?

Reply:
We always use the same offset for a cached index chunk. It is the offset of the first index entry that starts in the chunk.